### ISSUE TITLE
In train.py, the sample order between sample["advantages"] and sample["timesteps", "latents", "next_latents", "log_probs"] does not match.

### ISSUE THREAD

**Issue by YangSun22** [kvablack/ddpo-pytorch#29]
In train.py, the sample order between sample["advantages"] and sample["timesteps", "latents", "next_latents", "log_probs"] does not match.

```
# shuffle along time dimension independently for each sample
perms = torch.stack(
    [
        torch.randperm(num_timesteps, device=accelerator.device)
        for _ in range(total_batch_size)
    ]
)
for key in ["timesteps", "latents", "next_latents", "log_probs"]:
    samples[key] = samples[key][
        torch.arange(total_batch_size, device=accelerator.device)[:, None],
        perms,
    ]
```

advantages do not random with others, but they compute loss ?
```
# ppo logic
advantages = torch.clamp(
    sample["advantages"],
    -config.train.adv_clip_max,
    config.train.adv_clip_max,
)
ratio = torch.exp(log_prob - sample["log_probs"][:, j])
unclipped_loss = -advantages * ratio
clipped_loss = -advantages * torch.clamp(
    ratio,
    1.0 - config.train.clip_range,
    1.0 + config.train.clip_range,
)
loss = torch.mean(torch.maximum(unclipped_loss, clipped_loss))
```
